core/src/main/scala/clue/GraphQLException.scala

```diff
@@ -3,32 +3,15 @@
 
 package clue
 
-import cats.syntax.traverse._
-import clue.model.GraphQLError
-import clue.model.json.DecoderGraphQLError
-import io.circe.Decoder
-import io.circe.Json
+import clue.model.GraphQLErrors
 
 class GraphQLException(msg: String) extends Exception(msg)
 
-class ConnectionException() extends GraphQLException("Could not establish connection")
+case object ConnectionException extends GraphQLException("Could not establish connection")
 
-class DisconnectedException() extends GraphQLException("Connection was closed")
+case object DisconnectedException extends GraphQLException("Connection was closed")
```

Comment:
`Exception`s that are `case object`s can extend `scala.util.control.NoStackTrace` since their stack trace would be confusing since it's not related to the error.

Reply:
Ah good catch, maybe they should't be objects then, stack traces may be useful.